lib/suse/connect/client.rb

```diff
@@ -25,10 +26,43 @@ def initialize(config)
       def register!
         announce_or_update
         product = @config.product || Zypper.base_product
+
+        register_product(product, @config.product ? true : false)
+
+        # Only install recommended packages for base products
+        if product.isbase
+          product_hash = show_product(product)
+          product_hash.recommended_extensions.each do |ext|
+            register_product(ext)
+            register_recommended!(ext)
+          end
```

Comment:
Just call `register_recommended!(product)` instead? It does exactly the same thing.

Reply:
true. Good catch